packages/framework/presence/src/datastorePresenceManagerFactory.ts

```diff
@@ -26,7 +38,12 @@ class PresenceManagerDataObject extends LoadableFluidObject {
 		if (!this._presenceManager) {
 			// TODO: investigate if ContainerExtensionStore (path-based address routing for
 			// Signals) is readily detectable here and use that presence manager directly.
-			this._presenceManager = createPresenceManager(this.runtime);
+			const manager = createPresenceManager(this.runtime);
+			this.runtime.on("signal", (message: IInboundSignalMessage, local: boolean) => {
```

Comment:
Curious on why listening on events here? For a second I thought you are trying to decouple it from runtime, but runtime is still passed to presence manager.

Reply:
ok nevermind, runtime is not passed through.

Reply:
This is a move in that direction (to remove `IFluidDataStoreRuntime` from use). Eventually only `IExtensionRuntime` will be this package's connection to container runtime (`IExtensionRuntime` will probably be a subset of `IContainerRuntime`). But until container extension feature is implemented we are using `IFluidDataStoreRuntime`. One pattern that is different between connected as DataObject with `IFluidDataStoreRuntime` and connected as container extension is how inbound signals are sent. For container extension pattern, the extension [optionally] provides `processSignal` (see `PresenceExtensionInterface`). Here were are adapting DataObject to look more like the container extension host. So:
- listen to "signal" event here and call manager's `processSignal`
- do pass `IFluidDataStoreRuntime` down to manager but as `IEphemeralRuntime` which is the local definition of all of the things it needs from `IExtensionRuntime` _eventually_. You won't see `IExtensionRuntime` in `IEphemeralRuntime` because `IExtensionRuntime` is so unbaked as are the actual needs of presence. `IContainerRuntime` is the temporary proxy for `IExtensionRuntime`.

Super clear, right? We made a number of compromises to be able to get the basic functionality out and work with system as it is. This entire file will eventually disappear.